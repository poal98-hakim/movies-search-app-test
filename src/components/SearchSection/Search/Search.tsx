import React from "react";
import Button from "../../Button/Button";
import NumResults from "./NumResults/NumResults";

interface IProps{
  totalResults: number | null;
}
const Search : React.VFC<IProps> = ({ totalResults }) => {

  return (
    <div className="flex f-align-center">
        <Button type="submit" value="Search" />
        {totalResults ? <NumResults totalResults={totalResults} /> : null}
    </div>
  );
}

export default Search;
