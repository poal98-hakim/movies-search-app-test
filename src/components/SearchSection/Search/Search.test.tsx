import Search from "./Search";
import { render, screen } from '@testing-library/react';

const PROPS = {
  totalResults: 5
}

describe('Search', () => {
  let props = {...PROPS};

  beforeEach(() => { //before each test render the component and reset the value of props
    props = {...PROPS};
    render(<Search {...props} />);
  });
  
  it('renders correctly', () => {
    const { baseElement } = render(<Search {...props} />);
    expect(baseElement).toMatchSnapshot()
  })

  it('loads total results', () => {
    expect(screen.getByText(`${props.totalResults} results`)).toBeInTheDocument();
  })

  it('loads button', () => {
    expect(screen.getByRole("button")).toBeInTheDocument();
    expect(screen.getByText("Search")).toBeInTheDocument();
  })

})
