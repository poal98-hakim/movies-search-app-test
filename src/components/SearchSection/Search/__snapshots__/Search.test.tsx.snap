// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<body>
  <div>
    <div
      class="flex f-align-center"
    >
      <input
        class="button"
        type="submit"
        value="Search"
      />
      <span
        class="num-results-movies"
      >
        5
         results
      </span>
    </div>
  </div>
  <div>
    <div
      class="flex f-align-center"
    >
      <input
        class="button"
        type="submit"
        value="Search"
      />
      <span
        class="num-results-movies"
      >
        5
         results
      </span>
    </div>
  </div>
</body>
`;
